py-polars/polars/lazyframe/frame.py

```diff
@@ -1984,7 +1984,9 @@ def collect(
                 error_msg = f"collect() got an unexpected keyword argument '{k}'"
                 raise TypeError(error_msg)
 
-        new_streaming = _kwargs.get("new_streaming", False)
+        new_streaming = (
```

Comment:
Can we set the type of the `engine` argument to `engine: None | EngineType` and then:

```
if engine is None:
    engine = get_engine_affinity()
```

Reply:
I think `get_engine_affinity()` can still be `None`, so I set to "cpu" in that case.

Reply:
The return type on the rust side is `String`, so I don't think it can be `None` in python (I suggested removing the `Option` wrapper from the pyo3 wrapping function).

Reply:
Question: whose job should it be to validate that the provided engine affinity value is valid?

Reply:
It should be checked in the config I think.